pkg/handlers/nfd/handler.go

```diff
@@ -143,11 +146,7 @@ func (n *DefaultNFD) ensureNFDConfigmapForCluster(
 	}
 	err = client.ServerSideApply(ctx, n.client, cmForCluster)
 	if err != nil {
-		return nil, fmt.Errorf("failed to server side apply %w", err)
-	}
-	err = n.client.Get(ctx, ctrlclient.ObjectKeyFromObject(cmForCluster), cmForCluster)
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch server side applied configmap %w", err)
+		return nil, fmt.Errorf("failed to apply NFD ConfigMap for cluster: %w", err)
```

Comment:
Interesting, we don't need to get a fresh copy after the server side apply?

Reply:
Oh wait sorry no you don't need to refetch it, it's updated by the `ServerSideApply` call. Ultimately this calls `client.Patch` which is documented as `Patch patches the given obj in the Kubernetes cluster. obj must be a struct pointer so that obj can be updated with the content returned by the Server` so following a patch the passed in object is updated.

Reply:
This is the same semantics as `Create` or `Update`, where the passed in object is used in the request and updated with the response before returning to the caller.

Reply:
Awesome! Good change then. thank you